pkgs/by-name/gh/ghostty/package.nix

```diff
@@ -1,153 +1,207 @@
 {
+  lib,
+  stdenv,
   bzip2,
   callPackage,
-  expat,
   fetchFromGitHub,
   fontconfig,
   freetype,
   glib,
   glslang,
   harfbuzz,
-  lib,
-  libadwaita,
   libGL,
-  libpng,
   libX11,
-  libXcursor,
-  libXi,
-  libXrandr,
+  libadwaita,
   ncurses,
   nixosTests,
   oniguruma,
   pandoc,
   pkg-config,
   removeReferencesTo,
-  stdenv,
   versionCheckHook,
   wrapGAppsHook4,
   zig_0_13,
-  zlib,
+  # Usually you would override `zig.hook` with this, but we do that internally
+  # since upstream recommends a non-default level
+  # https://github.com/ghostty-org/ghostty/blob/4b4d4062dfed7b37424c7210d1230242c709e990/PACKAGING.md#build-options
+  optimizeLevel ? "ReleaseFast",
+  # https://github.com/ghostty-org/ghostty/blob/4b4d4062dfed7b37424c7210d1230242c709e990/build.zig#L106
+  withAdwaita ? true,
 }:
+
 let
-  # Ghostty needs to be built with --release=fast, --release=debug and
-  # --release=safe enable too many runtime safety checks.
   zig_hook = zig_0_13.hook.overrideAttrs {
-    zig_default_flags = "-Dcpu=baseline -Doptimize=ReleaseFast --color off";
+    zig_default_flags = "-Dcpu=baseline -Doptimize=${optimizeLevel} --color off";
   };
+
+  # https://github.com/ghostty-org/ghostty/blob/4b4d4062dfed7b37424c7210d1230242c709e990/src/apprt.zig#L72-L76
+  appRuntime = if stdenv.hostPlatform.isLinux then "gtk" else "none";
+  # https://github.com/ghostty-org/ghostty/blob/4b4d4062dfed7b37424c7210d1230242c709e990/src/font/main.zig#L94
+  fontBackend = if stdenv.hostPlatform.isDarwin then "coretext" else "fontconfig_freetype";
+  # https://github.com/ghostty-org/ghostty/blob/4b4d4062dfed7b37424c7210d1230242c709e990/src/renderer.zig#L51-L52
+  renderer = if stdenv.hostPlatform.isDarwin then "metal" else "opengl";
 in
+
 stdenv.mkDerivation (finalAttrs: {
   pname = "ghostty";
   version = "1.0.0";
 
+  outputs = [
+    "out"
+    "man"
+    "shell_integration"
+    "terminfo"
+    "vim"
+  ];
+
   src = fetchFromGitHub {
     owner = "ghostty-org";
     repo = "ghostty";
     tag = "v${finalAttrs.version}";
     hash = "sha256-AHI1Z4mfgXkNwQA8xYq4tS0/BARbHL7gQUT41vCxQTM=";
   };
 
-  strictDeps = true;
-
-  nativeBuildInputs = [
-    glib # Required for `glib-compile-schemas`
-    ncurses
-    pandoc
-    pkg-config
-    removeReferencesTo
-    wrapGAppsHook4
-    zig_hook
-  ];
-
-  buildInputs = [
-    bzip2
-    expat
-    fontconfig
-    freetype
-    glslang
-    harfbuzz
-    libadwaita
-    libGL
-    libpng
-    libX11
-    libXcursor
-    libXi
-    libXrandr
-    oniguruma
-    zlib
-  ];
-
-  dontConfigure = true;
-  # doCheck is set to false because unit tests currently fail inside the Nix sandbox.
-  doCheck = false;
-  doInstallCheck = true;
+  # Avoid using runtime hacks to help find X11
+  postPatch = lib.optionalString (appRuntime == "gtk") ''
+    substituteInPlace src/apprt/gtk/x11.zig \
+      --replace-warn 'std.DynLib.open("libX11.so");' 'std.DynLib.open("${lib.getLib libX11}/lib/libX11.so");'
```

Comment:
Out of curiosity: any idea why this zig lookup doesn't succeed without forcing the full path?

Reply:
Not really sure tbh. It could be the dlopen implementation internally; usually forcing linkage at runtime via `-l` would be enough. I also tried using `-rpath` before this, but no dice

Won't be a problem once https://github.com/ghostty-org/ghostty/pull/3857 gets into a stable tag though ;)